File: C's/cpp/1714C.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int t; cin >> t;

    while (t--) {
        int s; cin >> s;

        int n {10};
        int sum {0};
        vector<int> nums;
        while (n--) {
            if (sum < s) {
                if (sum + n <= s) {
                    sum += n;
                    nums.insert(nums.begin(), n);
                }
            }
        }

        for (auto c : nums) {
            cout << c;
        }
        cout << endl;
    }
    
    return 0;
}

File: C's/py/1714C.py
def main() -> None:
    t: int = int(input())

    while t:
        s: int = int(input())

        n: int = 9
        sum: int = 0
        nums: list[int] = []

        while n:
            if sum < s:
                if sum + n <= s:
                    sum += n
                    nums.insert(0, n)
            n -= 1

        print("".join(map(str, nums)))

        t -= 1

if __name__ == "__main__":
    main()